apps/aecore/src/aec_parent_connector.erl

```diff
@@ -415,30 +420,30 @@ get_pinning_data() ->
     end.
 
 get_generations(FromHeight, ToHeight) ->
-    ReversedBlocks =
-        lists:foldl(
-            fun(Height, Accum) ->
-                case aec_chain:get_generation_by_height(Height, backward) of
-                    {ok, #{micro_blocks := MBs}} ->
-                        ReversedGeneration = lists:reverse(MBs),
-                        ReversedGeneration ++ Accum;
-                    error -> error({failed_to_fetch_generation, Height})
-                end
-            end,
-            [],
-            lists:seq(FromHeight, ToHeight)),
-    lists:reverse(ReversedBlocks).
+    lists:flatten(
+        lists:map(
+        fun(Height) ->
+            case aec_chain:get_generation_by_height(Height, backward) of
+                {ok, #{micro_blocks := MBs}} -> MBs;
+                error -> error({failed_to_fetch_generation, Height})
+            end
+        end,
+        lists:seq(FromHeight, ToHeight)
+    )).
```

Comment:
Won't this make exactly as many calls to `get_generation_by_height` as the previous version?

Reply:
Yes it will...

You need something like (not tested):
```
get_micro_blocks_between(From, To) ->
    {ok, KeyHdr} = aec_chain:get_key_header_by_height(To),
    get_micro_blocks_between(From, KeyHdr, []).

get_micro_blocks_between(Stop, KeyHdr, MBs) ->
    case Stop > aec_headers:height(KeyHdr) of
        true -> lists:reverse(MBs);
        false ->
            {ok, PrevKeyHdr} = aec_chain:get_header(aec_headers:prev_key_hash(KeyHdr)),
            case aec_headers:prev_hash(KeyHdr) == aec_headers:prev_key_hash(KeyHdr) of
                true -> %% No Mb
                    get_micro_blocks_between(Stop, PrevKeyHdr, MBs);
                false ->
                    MB = aec_chain:get_block(aec_header:prev_hash(KeyHdr)),
                    get_micro_blocks_between(Stop, PrevKeyHdr, [MB | MBs])
            end
    end.
```

Reply:
TBH, this should go into `aec_chain` instead.